android/beagle/src/main/java/br/com/zup/beagle/android/components/ListViewTwo.kt

```
@@ -0,0 +1,523 @@
+
+        // remove esse holder da lista de holders recém criados. A próxima vez que esse holder passar por aqui,
+        // saberemos que esse holder foi reciclado.
+        createdViewHolderList.remove(holder)
+    }
+
+    override fun onViewRecycled(holder: ContextViewHolderTwo) {
+        super.onViewRecycled(holder)
+
+        // Quando um item da lista saí da árvore de views, precisamos salvar o contexto de todas as views que possuem
+        // contexto. Esses contextos são salvos aqui, pois enquanto o item estiver visível na lista, o usuário pode
+        // interagir com algum contexto desse template.
+        if (holder.adapterPosition > -1) {
+            saveContexts(holder.itemView, adapterItems[holder.adapterPosition].childContextMap)
+        }
+    }
+
+    override fun onViewAttachedToWindow(holder: ContextViewHolderTwo) {
+        super.onViewAttachedToWindow(holder)
+
+        // Quando um item entra na árvore de views, verificamos se ele foi completamente inicializado. Isso é feito aqui
+        // novamente, pois caso o onInit de todos os componentes do template complete antes do item estar visível na
+        // lista, o adapterPosition do holder é -1 e o método resolveComponentFinishedOnInit não consegue executar a
+        // lógica completa.
+
+        // O que fazemos aqui é indicar que para determinada posição do adapter, a execução de todos os onInit dentro
+        // do template terminou.
+        if (!adapterItems[holder.adapterPosition].initialized) {
+            val holderIsCompletellyInitialized = onInitControl[holder]?.isEmpty() ?: true
+            if (holderIsCompletellyInitialized) {
+                holder.setIsRecyclable(true)
+                adapterItems[holder.adapterPosition].initialized = true
+            }
+        }
+    }
+
+    fun setList(list: List<Any>) {
+        try {
+            adapterItems = list.map { BeagleAdapterItem(it) }
+            notifyDataSetChanged()
+        } catch (e: Exception) {
```

<table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/7f3ca72e85c2da4c26a6defda85bec6c74cf5a8e/android/beagle/src/main/java/br/com/zup/beagle/android/components/ListViewTwo.kt#L412"></span>Empty catch block detected. If the exception can be safely ignored, name the exception according to one of the exemptions as per the configuration of this rule.</td>
    </tr>
  </tbody>
</table>